api/tests/test_asset_log.py

```diff
@@ -316,6 +338,126 @@ def test_authenticated_admin_user_get_of_asset_logs_invalid_filter(
+        date = datetime.strptime(data[0]["created_at"], "%Y-%m-%dT%H:%M:%S.%fZ")
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(date.month, mock_datetime.return_value.month)
+        logs_count = AssetLog.objects.filter(
+            created_at__month=mock_datetime.return_value.month
+        ).count()
+        self.assertEqual(len(response.data["results"]), logs_count)
+
+    @patch("api.authentication.auth.verify_id_token")
+    def test_filter_asset_logs_by_invalid_month(self, mock_verify_id_token):
+        mock_verify_id_token.return_value = {"email": self.admin_user.email}
+        AssetLog.objects.create(
+            checked_by=self.security_user, asset=self.test_other_asset, log_type=CHECKIN
+        )
+        asset_logs_url = f"{self.asset_logs_url}?month=34"
+
+        response = client.get(
+            asset_logs_url, HTTP_AUTHORIZATION=f"Token {self.token_admin}"
+        )
+        self.assertEqual(response.status_code, 200)
+        logs_count = AssetLog.objects.filter(created_at__month="34").count()
+        self.assertEqual(len(response.data["results"]), logs_count)
+
+    @patch(
+        "django.utils.timezone.now",
+        return_value=datetime(2017, 9, 22, 17, 1, 26, 842_150),
+    )
+    @patch("api.authentication.auth.verify_id_token")
+    def test_filter_asset_logs_by_day(self, mock_verify_id_token, mock_datetime):
+        mock_verify_id_token.return_value = {"email": self.admin_user.email}
+        AssetLog.objects.create(
+            checked_by=self.security_user, asset=self.test_other_asset, log_type=CHECKIN
+        )
+        asset_logs_url = f"{self.asset_logs_url}?day={mock_datetime.return_value.day}"
+
+        response = client.get(
+            asset_logs_url, HTTP_AUTHORIZATION=f"Token {self.token_admin}"
+        )
+        data = response.data["results"]
+        date = datetime.strptime(data[0]["created_at"], "%Y-%m-%dT%H:%M:%S.%fZ")
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(date.day, mock_datetime.return_value.day)
+        logs_count = AssetLog.objects.filter(
+            created_at__day=mock_datetime.return_value.day
+        ).count()
+        self.assertEqual(len(response.data["results"]), logs_count)
+
+    @patch("api.authentication.auth.verify_id_token")
+    def test_filter_asset_logs_by_invalid_day(self, mock_verify_id_token):
+        mock_verify_id_token.return_value = {"email": self.admin_user.email}
+        AssetLog.objects.create(
+            checked_by=self.security_user, asset=self.test_other_asset, log_type=CHECKIN
+        )
+        asset_logs_url = f"{self.asset_logs_url}?day=456"
+
+        response = client.get(
+            asset_logs_url, HTTP_AUTHORIZATION=f"Token {self.token_admin}"
+        )
+        self.assertEqual(response.status_code, 200)
+        logs_count = AssetLog.objects.filter(created_at__day="456").count()
+        self.assertEqual(len(response.data["results"]), logs_count)
+
```

Comment:
one last thing 😅  
could we have one more test checking for the 3 attribute in the same fetch? eg `year=2019&month=5&day=15`